src/property_estimator_io.py

```diff
@@ -0,0 +1,731 @@
+            self.weights = {}
+            self.denominators = {}
+
+        def to_json(self):
+            """Converts this class into a JSON string.
+
+            Returns
+            -------
+            str
+                The JSON representation of this class.
+            """
+
+            value = {
+                'connection_options': self.connection_options.__getstate__(),
+                'estimation_options': self.estimation_options.__getstate__(),
+
+                'data_set_path': self.data_set_path,
+
+                'weights': {
+                    property_name: self.weights[property_name] for property_name in self.weights
+                },
+                'denominators': {
+                    property_name: self.denominators[property_name] for property_name in self.denominators
+                }
+            }
+
+            return json.dumps(value, sort_keys=True, indent=4, separators=(',', ': '), cls=TypedJSONEncoder)
+
+        @classmethod
+        def from_json(cls, json_source):
+            """Creates this class from a JSON string.
+
+            Parameters
+            -------
+            json_source: str or file-like object
+                The JSON representation of this class.
+            """
+
+            if isinstance(json_source, str):
+                with open(json_source, 'r') as file:
+                    dictionary = json.load(file, cls=TypedJSONDecoder)
+            else:
+                dictionary = json.load(json_source, cls=TypedJSONDecoder)
+
+            assert ('connection_options' in dictionary and
+                    'estimation_options' in dictionary and
+                    'data_set_path' in dictionary and
+                    'weights' in dictionary and
+                    'denominators' in dictionary)
+
+            value = cls()
+
+            value.connection_options = ConnectionOptions()
+            value.connection_options.__setstate__(dictionary['connection_options'])
+
+            value.estimation_options = PropertyEstimatorOptions()
+            value.estimation_options.__setstate__(dictionary['estimation_options'])
+
+            value.data_set_path = dictionary['data_set_path']
+
+            value.weights = {
+                property_name: dictionary['weights'][property_name] for property_name in dictionary['weights']
+            }
+            value.denominators = {
+                property_name: dictionary['denominators'][property_name] for property_name in dictionary['denominators']
+            }
+
+            return value
+
+    # A dictionary of the units that force balance expects each property in.
+    default_units = {
+        'Density': unit.kilogram / unit.meter ** 3,
+        'DielectricConstant': unit.dimensionless,
+        'EnthalpyOfVaporization': unit.kilojoules / unit.mole
+    }
+
+    def __init__(self, options, tgt_opts, forcefield):
+
+        super(PropertyEstimate_SMIRNOFF, self).__init__(options, tgt_opts, forcefield)
```

Comment:
(3/10 important. not blocking) Unless there's complex inheritance going on here, Python 3 supports (and recommends) using `super().__init__(...)` in this situation.
https://stackoverflow.com/a/576183

Reply:
I agree. Although we are writing new codes in python 3, I think we haven't officially removed python 2 support in ForceBalance yet, so I think this is good for now.